src/sentry/api/serializers/models/group.py

```diff
@@ -974,6 +1014,32 @@ def _execute_perf_seen_stats_query(
             referrer="serializers.GroupSerializerSnuba._execute_perf_seen_stats_query",
         )
 
+    @staticmethod
+    def _execute_generic_seen_stats_query(
+        item_list, start=None, end=None, conditions=None, environment_ids=None
+    ):
+        project_ids = list({item.project_id for item in item_list})
+        group_ids = [item.id for item in item_list]
+        aggregations = [
```

Comment:
It would be nice to see these new queries be written using SnQL instead of the legacy format.

Reply:
Yeah, that's definitely something we want to do since SnQL, from what I can skim through, is a lot more intuitive to use. Unfortunately, there's a lot of baked in logic in the query processing tied to the legacy querying that would make porting the lagacy format to SnQL a bit unwieldy for this PR. 

We'll keep this in mind and start using SnQL for future queries and I'll try backporting this query when we don't have immediate deadlines to hit.

Reply:
Not to beat a dead horse, but you're building a full query here and calling `aliased_query`. Couldn't you build a SnQL query instead and call `raw_snql_query`?

Reply:
I think my only reservations about directly using SnQL versus legacy is the accumulated processing here:
https://github.com/getsentry/sentry/blob/e36ddcf9e657c22818df301e525494357777a185/src/sentry/utils/snuba.py#L522